### DISCUSSION TITLE
Job output cannot be using when triggering workflow from job

### DISCUSSION THREAD

**Discussion by afonsoc12** [community/community#26664] [Actions]
<p>Hi,</p>
<p>So I have this workflow that gathers the version from a website and then triggers a build (another workflow), via the workflow_call trigger. This is the code of the scheduler:</p>
<pre><code class="lang-auto">name: Checks for new releases and deploys master
on:
  schedule:
      - cron: '15 */2 * * *'
  workflow_dispatch:

jobs:
  external_trigger:
    runs-on: ubuntu-latest
    outputs:
      version: ${{ steps.version_getter.outputs.version }}
    steps:
      - uses: actions/checkout@v2.3.3
      
      - name: Version diff
        run: |
          chmod u+x ./scripts/isUpdated.sh
          ./scripts/isUpdated.sh

      - id: version_getter
        name: Trigger build check
        if: ${{ hashFiles('DOCKER_VERSION.txt') != '' }}
        run: |
          version=$(cat DOCKER_VERSION.txt)
          echo "Triggering because there is the new version $version"
          echo "::set-output name=version::$version"
          
  trigger:
    needs: external_trigger
    uses: ./.github/workflows/docker_build_master.yml
    with:
      version: ${{ needs.external_trigger.outputs.version }}

</code></pre>
<p>The problem is that the version variable is being set as empty string. I have tried using a step in the last job and it works, its just when I use the <code>uses</code> to trigger the other workflow that the output does not work. Like this:</p>
<pre><code class="lang-auto">trigger:
    needs: external_trigger
    runs-on: ubuntu-latest
    steps:
       - run: echo "${{ needs.external_trigger.outputs.version }}"
</code></pre>
<p>Any fix for this issue?<br>
Thanks</p>

**Comment by jsoref** [comment]
<p>Afaict, from:<br>
<a href="https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability" class="onebox" target="_blank" rel="noopener">https://docs.github.com/en/actions/learn-github-actions/contexts#context-availability</a></p>
<div class="md-table">
<table>
<thead>
<tr>
<th>Workflow key</th>
<th>Context</th>
<th>Special functions</th>
</tr>
</thead>
<tbody>
<tr>
<td><code>jobs.&lt;job_id&gt;.steps.with</code></td>
<td><code>github, needs, strategy, matrix, job, runner, env, secrets, steps, inputs</code></td>
<td><code>hashFiles</code></td>
</tr>
<tr>
<td><code>jobs.&lt;job_id&gt;.with.&lt;with_id&gt;</code></td>
<td><code>github, needs</code></td>
<td></td>
</tr>
</tbody>
</table>
</div><p><code>needs</code> should be available (and presumably work) in both cases…</p>
<p>Can you provide a sample repository?</p>

**Comment by afonsoc12** [comment]
<p>Sure! <a href="https://github.com/afonsoc12/docker-cloudflared/blob/06ee46a846e204fe5c0fd6a6fa3954b6120a3119/.github/workflows/scheduler_master.yml" rel="noopener nofollow ugc">Here</a> it is.</p>
<p>If you check the job trigger, the output is not propagated, but it works on job tst.<br>
Any thoughts?</p>

**Comment by jsoref** [comment]
<p>Personally, I’d want to fix this:</p><aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/afonsoc12/docker-cloudflared/runs/6476100269?check_suite_focus=true#step:3:8">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/afonsoc12/docker-cloudflared/runs/6476100269?check_suite_focus=true#step:3:8" target="_blank" rel="noopener">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/2119212/181099110-a582c6d2-98c9-447f-83dd-92e5db1ba441.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/c/dcc9ee4ab74f98f31f63805800c9a2df2a7868ac_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/d/c/dcc9ee4ab74f98f31f63805800c9a2df2a7868ac_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/d/c/dcc9ee4ab74f98f31f63805800c9a2df2a7868ac.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/c/dcc9ee4ab74f98f31f63805800c9a2df2a7868ac_2_10x10.png"></div>

<h3><a href="https://github.com/afonsoc12/docker-cloudflared/runs/6476100269?check_suite_focus=true#step:3:8" target="_blank" rel="noopener">Checks for new releases and deploys master ·...</a></h3>

  <p>Contribute to afonsoc12/docker-cloudflared development by creating an account on GitHub.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>I’d probably add a <code>set -x</code> to the set output step.</p>
<p>I’d probably also enable debug logging:<br>
<a href="https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging" class="onebox" target="_blank" rel="noopener">https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/enabling-debug-logging</a></p>

**Comment by afonsoc12** [comment]
<p>Thanks, sure the scripts are still very crude, I am looking forward to fix CI first.<br>
While the output is shown <a href="https://github.com/afonsoc12/docker-cloudflared/runs/6476542534?check_suite_focus=true#step:2:5" rel="noopener nofollow ugc">on this one</a>, it <a href="https://github.com/afonsoc12/docker-cloudflared/runs/6476542773?check_suite_focus=true#step:2:1" rel="noopener nofollow ugc">does not show here</a>, which is going to use another workflow</p>

**Comment by jsoref** [comment]
<p><a href="https://github.com/check-spelling/docker-cloudflared/runs/6478217149?check_suite_focus=true#step:3:1">https://github.com/check-spelling/docker-cloudflared/runs/6478217149?check_suite_focus=true#step:3:1</a></p>

**Comment by jsoref** [comment, marked as answer]
<p>I should probably explain, but in short:<br>
<a href="https://github.com/afonsoc12/docker-cloudflared/commit/ee01e193f900aedac6850199d320f4b85fca4e7e" class="onebox" target="_blank" rel="noopener">https://github.com/afonsoc12/docker-cloudflared/commit/ee01e193f900aedac6850199d320f4b85fca4e7e</a></p>
<hr>
<p>A slightly longer explanation is certainly in order.</p>
<p>Literally, the google completion for <code>github.event.inputs</code> is <code>github.event.inputs vs inputs</code>.</p>
<p>GitHub supports events that have inputs:<br>
<a href="https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#providing-inputs" class="onebox" target="_blank" rel="noopener">https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#providing-inputs</a><br>
It handles that via <code>github.event.inputs</code></p>
<p>It also supports giving (reusable) workflows inputs:<br>
<a href="https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow" class="onebox" target="_blank" rel="noopener">https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-inputs-and-secrets-in-a-reusable-workflow</a></p>
<p>In that case, inputs come via <code>inputs</code></p>
<p>Note that this aligns with actions:<br>
<a href="https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs" class="onebox" target="_blank" rel="noopener">https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#inputs</a><br>
where they also come via <code>inputs</code>:<br>
<a href="https://docs.github.com/en/actions/learn-github-actions/contexts#inputs-context" class="onebox" target="_blank" rel="noopener">https://docs.github.com/en/actions/learn-github-actions/contexts#inputs-context</a></p>

**Reply by longyarnz** [reply]
I don't know why @afonsoc12 marked this comment as the answer but it doesn't answer the question.

**Reply by jsoref** [reply]
So, the trick is that there are two files that were supposed to work together, one generated some outputs and tried to pass those as inputs to the other. The problem *wasn't* in the file with the outputs, it was in the file trying to take those as inputs.

You can see how they debugged and fixed it here:
https://github.com/afonsoc12/docker-cloudflared/blob/f593a4d237b07cc7ce0a60130a7e46da250e34f7/.github/workflows/docker_build_master.yml#L25-L27
```sh
        echo "github.events.inputs: ${{ github.event.inputs.version }}"
        echo "inputs: ${{ inputs.version }}"
        echo "::set-output name=version::${{ github.event.inputs.version || inputs.version }}"
```

I'm sorry that the answer didn't clarify that the question was misstated, but, until someone understands what's going wrong, all they can do is describe what they think they see -- and they thought their output wasn't getting to the next stage, and they were kinda right -- so what they wrote as a question was understandable.

**Comment by afonsoc12** [comment]
<p>Hi,<br>
Thanks this was massive help! I’ll read those links closely</p>

**Comment by gatsbyz** [comment]
> The problem is that the version variable is being set as empty string.

This problem still persists. 

```
trigger:
  needs: external_trigger
  uses: ./.github/workflows/docker_build_master.yml
  with:
    version: ${{ needs.external_trigger.outputs.version }}
```
with `uses:`, `needs.job_id.outputs.xxxxx` is returning an empty string

seems like OP's solution was to just paste the `uses:` workflow into the same file. https://github.com/afonsoc12/docker-cloudflared/blob/master/.github/workflows/docker_build_master.yml#L29-L39

**Comment by khiemns54** [comment]
I am still facing the similar issue.
